### PR TITLE
Uses assemblies in docker images and enables arm64+amd64 platforms

### PR DESCRIPTION
**Pull request by mattjohnson**
Changes included in this PR:
- Support both linux/amd64 and linux/arm64 architectures
- Update base images to use ubi-minimal
- Use assemblies that contain embedded JDK instead of using openjdk base image (includes too many unnecessary packages)
- Sets up docker trust for manifest list tags
- Builds all architectures on branch jobs

Overall docker image size is reduced by about 300MB.

### CODE REVIEW

File: Jenkinsfile.release [hunk 1]
```diff
@@ -123,17 +121,16 @@ void pushImage(String imageName) {
...
-
-      withCredentials([string(credentialsId: 'sonatype-password', variable: 'DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE')]) {
-        runSafely "docker image push ${imageName}:${env.VERSION}"
-        runSafely "docker image push ${imageName}:latest"
+        sh  """
+        curl -L https://go.dev/dl/go1.23.3.linux-amd64.tar.gz | tar -xzf -
+        export PATH=${env.PATH}:${env.WORKSPACE}/go/bin:${env.WORKSPACE}/bin
+        export GOPATH=${env.WORKSPACE}
+        go install -tags pkcs11 github.com/theupdateframework/notary/cmd/notary@latest
+        notary --help
```
Comment (eduard-tita) [inline Jenkinsfile.release:130]: `notary --help` - Is this line needed here?

File: Jenkinsfile.slim.release [hunk 2]
```diff
@@ -117,17 +115,16 @@ void pushImage(String imageName) {
...
-
-      withCredentials([string(credentialsId: 'sonatype-password', variable: 'DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE')]) {
-        runSafely "docker image push ${imageName}:${env.VERSION}-slim"
-        runSafely "docker image push ${imageName}:latest-slim"
+        sh  """
+        curl -L https://go.dev/dl/go1.23.3.linux-amd64.tar.gz | tar -xzf -
+        export PATH=${env.PATH}:${env.WORKSPACE}/go/bin:${env.WORKSPACE}/bin
+        export GOPATH=${env.WORKSPACE}
+        go install -tags pkcs11 github.com/theupdateframework/notary/cmd/notary@latest
+        notary --help
```
Comment (eduard-tita) [inline Jenkinsfile.slim.release:124]: Same as above - is this line needed here?

**Review by eduard-tita — APPROVED** [review]
+0.5 Nothing stands out to me as wrong, but I cannot say I fully understand all the details. 

Especially the notarization approach is new to me, so I cannot tell by just looking at the PR if it works or not, or if anything is missing.

File: Jenkinsfile.release (hunk 1 above)
Comment (mattjohnson) [reply Jenkinsfile.release:130]: Not strictly needed. I liked it because it helped me debug locally to know it was installed and setup correctly.

**Pull request merged**